core/src/main/java/com/linecorp/armeria/internal/AnnotatedHttpServiceParamUtil.java

```diff
@@ -208,6 +217,135 @@ public static Object convertParameter(@Nullable String value,
+        if (param == null) {
+            return null;
+        }
+
+        final String value = param.value();
+        if (DefaultValues.isSpecified(value)) {
+            checkArgument(!value.trim().isEmpty(), "value: %s (expected: not blank)");
+            return value;
+        }
+        return getName(nameRetrievalTarget);
+    }
+
+    /**
+     * Returns the value of the {@link Header} annotation which is specified on the {@code element} if
+     * the value is not blank. If the value is blank, it returns the name of the specified
+     * {@code element} object which is an instance of {@link Parameter} or {@link Field}.
+     *
+     * <p>Note that the name of the specified {@code element} will be converted as
+     * {@link CaseFormat#LOWER_HYPHEN} that the string elements are separated with one hyphen({@code -})
+     * character. The value of the {@link Header} annotation will not be converted because it is clearly
+     * specified by a user.
+     */
+    @Nullable
+    public static String findHeaderName(AnnotatedElement element) {
+        return findHeaderName(element, element);
+    }
+
+    /**
+     * Returns the value of the {@link Header} annotation which is specified on the {@code element} if
+     * the value is not blank. If the value is blank, it returns the name of the specified
+     * {@code nameRetrievalTarget} object which is an instance of {@link Parameter} or {@link Field}.
+     *
+     * <p>Note that the name of the specified {@code nameRetrievalTarget} will be converted as
+     * {@link CaseFormat#LOWER_HYPHEN} that the string elements are separated with one hyphen({@code -})
+     * character. The value of the {@link Header} annotation will not be converted because it is clearly
+     * specified by a user.
+     */
+    @Nullable
+    public static String findHeaderName(AnnotatedElement element, Object nameRetrievalTarget) {
+        requireNonNull(element, "element");
+        requireNonNull(nameRetrievalTarget, "nameRetrievalTarget");
+        final Header header = element.getAnnotation(Header.class);
+        if (header == null) {
+            return null;
+        }
+
+        final String value = header.value();
+        if (DefaultValues.isSpecified(value)) {
+            checkArgument(!value.trim().isEmpty(), "value: %s (expected: not blank)");
+            return value;
+        }
+        return toHeaderName(getName(nameRetrievalTarget));
+    }
+
+    /**
+     * Returns whether the specified {@code element} is annotated with a {@link Param} or {@link Header}.
+     */
+    public static boolean hasParamOrHeader(AnnotatedElement element) {
+        return element.isAnnotationPresent(Param.class) ||
+               element.isAnnotationPresent(Header.class);
+    }
+
+    /**
+     * Returns {@code true} if the specified {@code element} is not annotated with a {@link Param} nor
+     * {@link Header}.
+     */
+    public static boolean hasNeitherParamNorHeader(AnnotatedElement element) {
```

Comment:
It seems like this method is not used anywhere and the name is sort of confusing. How about removing this method?

Reply:
Replaced  `allMatch(AnnotatedHttpServiceParamUtil::hasNeitherParamNorHeader)` with `.noneMatch(AnnotatedHttpServiceParamUtil::hasParamOrHeader)` and removed this. Thanks!